updater.go

```diff
@@ -184,6 +182,77 @@ func (u *Updater) update(ctx Context, options UpdateOptions) (*Update, error) {
 	return update, nil
 }
 
+func (u *Updater) ApplyDownloaded(ctx Context) (bool, error) {
+	options := ctx.UpdateOptions()
+
+	// 1. check with the api server again for the latest update to be sure that a new update has not come out since our last call to CheckAndDownload
+	u.log.Infof("Attempting to apply previously downloaded update")
+	update, err := u.checkForUpdate(ctx, options)
+
+	if err != nil {
+		return false, findErr(err)
+	}
+
+	// Only report apply success/failure
+	applied, err := u.applyDownloaded(ctx, update, options)
+	if err != nil && !applied {
```

Comment:
applied should never be true if there was an error, just combine the report in the error check below

Reply:
The real issue here was `return false, nil` when I should have just formatted an error.